test_data/Zilien version 1/2022-04-06 16_17_23 full set.tsv

```diff
@@ -0,0 +1,850 @@
+num_header_lines	Number of header lines		int	115
```

Comment:
I wonder if it's worth shrinking the data in the file down, like you did for test_data/biologic/Pt_poly_cv_CUT.mpt?

Reply:
The thing is, when I cut the other one, it was out of concern for test time, but since all the data is now being read in through numpy.genfromtxt I don't think it will make much difference. I decided to measure it and the data supports it:
```text
…/ixdat zilien_parser +50-26 $» v3.9.5 (ixdat) 
at 11:41:28 ❯ pytest tests -k zilien_reader -vv --durations=0
================================================= test session starts ==================================================
platform linux -- Python 3.9.5, pytest-7.1.0, pluggy-1.0.0 -- /home/kenneth/venv/ixdat/bin/python
cachedir: .pytest_cache
rootdir: /home/kenneth/venv/ixdat/ixdat
plugins: profiling-1.7.0
collecting ... importing ixdat v0.2.1 from /home/kenneth/venv/ixdat/ixdat/src/ixdat/__init__.py
collected 93 items / 86 deselected / 7 selected                                                                        

tests/functional/test_zilien_reader.py::test_read[Measurement-expected_series0] PASSED                           [ 14%]
tests/functional/test_zilien_reader.py::test_read[ECMSMeasurement-expected_series1] PASSED                       [ 28%]
tests/functional/test_zilien_reader.py::test_read[MSMeasurement-expected_series2] PASSED                         [ 42%]
tests/functional/test_zilien_reader.py::test_read[ECMeasurement-expected_series3] PASSED                         [ 57%]
tests/functional/test_zilien_reader.py::test_read[EC-MS-expected_series4] PASSED                                 [ 71%]
tests/functional/test_zilien_reader.py::test_read[EC-expected_series5] PASSED                                    [ 85%]
tests/functional/test_zilien_reader.py::test_read[MS-expected_series6] PASSED                                    [100%]

================================================== slowest durations ===================================================
0.03s call     tests/functional/test_zilien_reader.py::test_read[Measurement-expected_series0]
0.01s call     tests/functional/test_zilien_reader.py::test_read[ECMSMeasurement-expected_series1]
0.01s call     tests/functional/test_zilien_reader.py::test_read[EC-expected_series5]
0.01s call     tests/functional/test_zilien_reader.py::test_read[EC-MS-expected_series4]
0.01s call     tests/functional/test_zilien_reader.py::test_read[MSMeasurement-expected_series2]
0.01s call     tests/functional/test_zilien_reader.py::test_read[MS-expected_series6]
0.01s call     tests/functional/test_zilien_reader.py::test_read[ECMeasurement-expected_series3]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[Measurement-expected_series0]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[ECMeasurement-expected_series3]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[EC-expected_series5]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[ECMSMeasurement-expected_series1]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[MS-expected_series6]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[EC-MS-expected_series4]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[MSMeasurement-expected_series2]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[ECMeasurement-expected_series3]
0.00s setup    tests/functional/test_zilien_reader.py::test_read[EC-expected_series5]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[Measurement-expected_series0]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[ECMSMeasurement-expected_series1]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[MSMeasurement-expected_series2]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[MS-expected_series6]
0.00s teardown tests/functional/test_zilien_reader.py::test_read[EC-MS-expected_series4]
=========================================== 7 passed, 86 deselected in 0.89s ===========================================
```
As you can see, actually executing all these tests amount to just 100ms. The rest of the 0.9s must be spent discovering the tests. The only other concern is archive size (on disk and in transfer), but here I think ~500kb per parser on disk and 115kb compressed in transfer is not an unreasonable amount, especially considering we will likely not have hundreds of parsers.

Reply:
got it, then it's fine :)